File: src/main/java/com/monkeyteam/monkeyutils/MonkeyUtils.java
package com.monkeyteam.monkeyutils;

import com.monkeyteam.monkeyutils.modules.*;
import org.bukkit.Bukkit;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;

public final class MonkeyUtils extends JavaPlugin {

    private static MonkeyUtils instance;

    FileConfiguration config = getConfig();

    @Override
    public void onEnable() {
        // Plugin startup logic
        instance = this;

        // Events
        getServer().getPluginManager().registerEvents(new CropsBreak(), this);
        getServer().getPluginManager().registerEvents(new Back(), this);
        getServer().getPluginManager().registerEvents(new MOTD(), this);
        getServer().getPluginManager().registerEvents(new Cat(), this);
        getServer().getPluginManager().registerEvents(new Whitelist(), this);

        // Commands
        this.getCommand("openinventory").setExecutor(new OpenInventory());
        this.getCommand("back").setExecutor(new Back());
        this.getCommand("home").setExecutor(new Home.HomeCommand());
        this.getCommand("sethome").setExecutor(new Home.SethomeCommand());
        this.getCommand("monkeyreload").setExecutor(new Reload());
        this.getCommand("whitelist").setExecutor(new Whitelist());

        Bukkit.getLogger().fine("Ready to monkeying !");
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }

    public static MonkeyUtils getInstance() {
        return instance;
    }
}

File: src/main/java/com/monkeyteam/monkeyutils/MonkeyUtils.kt
package com.monkeyteam.monkeyutils

import com.monkeyteam.monkeyutils.modules.*
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin

class MonkeyUtils : JavaPlugin() {

  companion object {
    private var realInstance: MonkeyUtils? = null
    val instance
      get() = realInstance!!

    fun log(message: String) {
      Bukkit.getConsoleSender().sendMessage("§6[MonkeyUtils] §r${message}")
    }
  }

  override fun onEnable() {
    realInstance = this

    // Events
    server.pluginManager.registerEvents(CropsBreak, this)
    server.pluginManager.registerEvents(Back, this)
    server.pluginManager.registerEvents(Cat, this)
    server.pluginManager.registerEvents(MOTD, this)
    server.pluginManager.registerEvents(Whitelist, this)

    // Commands
    getCommand("openinventory")!!.setExecutor(OpenInventory)
    getCommand("back")!!.setExecutor(Back)
    getCommand("home")!!.setExecutor(Home.HomeCommand)
    getCommand("sethome")!!.setExecutor(Home.SetHomeCommand)
    getCommand("monkeyreload")!!.setExecutor(Reload)
    getCommand("whitelist")!!.setExecutor(Whitelist)

    log("Ready to monkeying !")
  }

  override fun onDisable() {
    // Plugin shutdown logic
  }
}
